app/v6/retrieveForeignPropertyDetails/def1/model/response/EndReason.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v6.retrieveForeignPropertyDetails.def1.model.response
+
+import play.api.libs.json.Format
+import shared.utils.enums.Enums
+
+enum EndReason(val fromDownstream: String) {
+  case `no-longer-renting-property-out` extends EndReason("noLongerRentingPropertyOut")
+  case `disposal`                       extends EndReason("disposal")
+  case `added-in-error`                 extends EndReason("addedInError")
+  case `cessation`                      extends EndReason("cessation")
+}
+
+object EndReason {
+
+  given Format[EndReason]                        = Enums.format(values)
+  val parser: PartialFunction[String, EndReason] = Enums.parser[EndReason](values)
```

Comment:
Prefer this to be changed to 
```
given Reads[EndReason]  = Enums.readsFrom(values, _.fromDownstream)
given Writes[EndReason] = Enums.writes
```

The readsFrom method is not currently included in the repository but can be found [here](https://github.com/hmrc/self-assessment-accounts-api/blob/32c9aaba9015068eb927923e413c1d05a608f56a/app/shared/utils/enums/Enums.scala#L34). Make sure to include supporting tests with this method.

Then we can just use OFormat in `ForeignPropertyDetailsEntry`